src/utils/hooks/useLoadApplicationNEW.ts

```diff
@@ -13,34 +13,56 @@ import { buildSectionsStructure } from '../helpers/structure/buildSectionsStruct
 import {
   Application,
   ApplicationSection,
+  ApplicationStageStatusAll,
+  ApplicationStatus,
   Template,
   TemplateElement,
   TemplateStage,
   useGetApplicationNewQuery,
 } from '../generated/graphql'
 
+const MAX_REFETCH = 10
+
 const useLoadApplication = ({ serialNumber, networkFetch }: UseGetApplicationProps) => {
   const [isLoading, setIsLoading] = useState(true)
   const [structureError, setStructureError] = useState('')
   const [structure, setFullStructure] = useState<FullStructure>()
   const [template, setTemplate] = useState<TemplateDetails>()
+  const [refetchAttempts, setRefetchAttempts] = useState(0)
 
-  const { data, error } = useGetApplicationNewQuery({
+  const { data, loading, error, refetch } = useGetApplicationNewQuery({
     variables: {
       serial: serialNumber,
     },
     fetchPolicy: networkFetch ? 'network-only' : 'cache-first',
+    notifyOnNetworkStatusChange: true,
   })
 
   useEffect(() => {
-    if (!data) return
+    if (!data || loading) return
     const application = data.applicationBySerial as Application
 
-    if (!application) {
+    if (!application || !application.template) setStructureError('No application/template found')
+    if (!data.applicationStageStatusAlls?.nodes) setStructureError('No status found')
+
+    if (structureError) {
       setIsLoading(false)
       return
     }
 
+    // Checking if trigger is running before loading current status
+    if (data.applicationTriggerStates?.nodes) {
+      const triggerResult = data.applicationTriggerStates?.nodes
+      if (!triggerResult.every((result) => result === null) && refetchAttempts < MAX_REFETCH) {
```

Comment:
since MAX_REFETCH is being implemented.

* if max refetch is reached then should check it's been reached and set error with return 
*  after successfully getting application with trigger == NULL set refetch attempts to 0

Reply:
Done: 13b7b1c